common_benchmark_suite/openxla/benchmark/models/tf/efficientnet/efficientnet_model.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2023 The OpenXLA Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import tensorflow as tf
+
+from typing import Any, Tuple
+from openxla.benchmark.models import model_interfaces, utils
+
+DEFAULT_IMAGE_URL = "https://storage.googleapis.com/iree-model-artifacts/ILSVRC2012_val_00000023.JPEG"
+
+# EfficientNet uses different image input sizes depending on the variant.
+# The larger the model, the larger the image size.
+MODEL_NAME_TO_INPUT_SIZE = {
+    "efficientnetb0": 224,
+    "efficientnetb1": 240,
+    "efficientnetb2": 260,
+    "efficientnetb3": 300,
+    "efficientnetb4": 380,
+    "efficientnetb5": 456,
+    "efficientnetb6": 528,
+    "efficientnetb7": 600,
+}
+
+MODEL_NAME_TO_CLASS = {
+    "efficientnetb0": tf.keras.applications.efficientnet.EfficientNetB0,
+    "efficientnetb1": tf.keras.applications.efficientnet.EfficientNetB1,
+    "efficientnetb2": tf.keras.applications.efficientnet.EfficientNetB2,
+    "efficientnetb3": tf.keras.applications.efficientnet.EfficientNetB3,
+    "efficientnetb4": tf.keras.applications.efficientnet.EfficientNetB4,
+    "efficientnetb5": tf.keras.applications.efficientnet.EfficientNetB5,
+    "efficientnetb6": tf.keras.applications.efficientnet.EfficientNetB6,
+    "efficientnetb7": tf.keras.applications.efficientnet.EfficientNetB7,
+}
+
+
+class EfficientNet(model_interfaces.InferenceModel, tf.Module):
+  """See https://huggingface.co/docs/transformers/model_doc/resnet for more information."""
+
+  batch_size: int
+  model: Any
+  model_name: str
+  input_size: int
+
+  def __init__(self, batch_size: int, model_name: str):
+    self.batch_size = batch_size
+    self.model_name = model_name
+    self.input_size = MODEL_NAME_TO_INPUT_SIZE[model_name]
+    self.model = MODEL_NAME_TO_CLASS[model_name]()
+
+  def generate_default_inputs(self) -> Tuple[Any, ...]:
+    # TODO(#44): This should go away once we support different raw inputs.
+    image = utils.download_and_read_img(DEFAULT_IMAGE_URL,
+                                        width=self.input_size,
+                                        height=self.input_size)
```

Comment:
I feel like we should resize the image in `preprocess` step. This way, we can use the same raw input image for models required different image sizes? (when we support non-default raw inputs)

Reply:
That makes sense. I moved the resizing to the preprocessing.